core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -378,6 +378,118 @@ object TestUtils extends Logging {
     props
   }
 
+  def createCombinedControllerConfig(nodeId: Int,
```

Comment:
Can't we just add a boolean to `TestUtils.createBrokerConfig`? Two whole new functions seems like overkill.

Also, while you're changing this code, it would be good to use the standard indentation style. Having this be indented halfway through the page really hurts the eyes. Thsee days we do:

```
def myVeryLongFunctionName(
  foo: Foo,
  bar: Bar,
...
  quux: Quux
): ReturnValueType = {
  ...
}
```